src/common/guided_filter.c

```diff
@@ -323,8 +323,10 @@ static int _cl_split_rgb(const int devid,
                           cl_mem imgg_b,
                           const float guide_weight)
 {
+  const int clwidth = ROUNDUPDWD(width, devid);
+  const int clheight = ROUNDUPDHT(height, devid);
   const int kernel = darktable.opencl->guided_filter->kernel_guided_filter_split_rgb;
-  return dt_opencl_enqueue_kernel_2d_args(devid, kernel, width, height,
+  return dt_opencl_enqueue_kernel_2d_args(devid, kernel, clwidth, clheight,
```

Comment:
I'm a bit confused by this as  `dt_opencl_enqueue_kernel_2d_args` / `dt_opencl_enqueue_kernel_2d_args_internal` already contains (since a large deduplication effort a while back)
```
  const size_t sizes[] = { ROUNDUPDWD(w, dev), ROUNDUPDHT(h, dev), 1 };
```
Presumably rounding up twice should not change anything? This looks like a dededup to me :-)

Reply:
It is and i already spotted that too.